.circleci/config.yml

```diff
@@ -10,18 +10,33 @@ commands:
       env:
         default: "py36"
         type: string
+      coveralls_token:
+        default: ""
+        type: string
     steps:
-      - run: tox --recreate -e <<parameters.env>>
-
+      - run:
+          name: run tox
+          command: |
+            echo "export COVERALLS_REPO_TOKEN=<<parameters.coveralls_token>>" >> $BASH_ENV
+            source $BASH_ENV
+            tox --recreate -e <<parameters.env>>
```

Comment:
An easier and more recommended way that doesn't affect the environment for subsequent commands is to do the following, assuming circleci just passes the command to the shell directly:

`COVERALLS_REPO_TOKEN=<<parameters.coveralls_token>> tox --recreate -e <<parameters.env>>`

Obviously this won't work if the token is used later, in which case ignore this comment. :-)

Reply:
You are right. That is much clearer and exactly what I need here. The env var is only used to run Tox so it is perfect.
I'm pretty sure I tried a similar approach using directly the env var and it didn't work, the env var was not set to the context when calling tox. That is why I ended up using the parameter here. 
I did not think about using the parameter directly as you suggested. I'm glad you noticed that because it looks much better now.
 
Thanks Scott.